### ISSUE TITLE
Syntactic shortcut to pipe output into a parameter that doesn't support pipeline input

### ISSUE THREAD

**Issue by dbaileyut** [PowerShell/PowerShell#18886]
### Summary of the new feature / enhancement

I often find cmdlets that have limited support for parameters with pipeline input. Currently, I do something like the following:
`Get-Thing | Foreach-Object { Set-OtherThing -UnsupportedParam $_ }`

It would be cool if this could be expressed more concisely. Note: I'm mostly concerned about command line usage here rather than scripting.

### Proposed technical implementation details (optional)

I'm not sure of the cleanest way to do this but off the top of my head it might look like:
`Get-Thing | Set-OtherThing -UnsupportedParam %{$_}`

**Comment by mklement0** [comment]
Note that if you use `%` in lieu of `ForEach-Object`, you get a solution that is as concise as what you propose, except that the symbols are arranged diffrently:

```powershell
 Get-Thing | % { Set-OtherThing -UnsupportedParam $_ }
```

However, I can see the appeal of the symmetry with [delay-bind script blocks](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_script_blocks#using-delay-bind-script-blocks-with-parameters), which would imply the following syntax:

```powershell
# Currently only works with pipeline-binding cmdlets in general, and the specific parameter targeted
# with the script block, -UsupportedParam, must *also* be declared as pipeline-binding.
# Additionally, the parameter type must neither be a [scriptblock] nor [object] / untyped.
Get-Thing | Set-OtherThing -UnsupportedParam { $_ }
```

However, there are challenges:

* Conceptual: Someone reading the code won't be able to tell whether the command of interest is truly pipeline-binding or not, which notably makes a difference performance-wise, given that a non-pipeline-binding one must be called _once for each input object_ - which is what `Foreach-Object { Set-OtherThing -UnsupportedParam $_ }` makes obvious.

* Backward compatibility: At least at first glance it seems there _wouldn't_ be a concern, though that needs to be carefully evaluated:  
  * It is the presence of a script-block argument passed to a non-pipeline-binding parameter that is neither `[scriptblock]` nor `[object]`-typed / untyped that would trigger the new behavior, and passing a script block to an existing non-pipeline-binding command's parameters with those type restrictions would currently _break_.
  * As an aside: _Simple_ functions or scripts (those with neither a `[CmdletBinding()]` nor a `[Parameter()]` attribute) can _also_ accept pipeline input, even if not formally declared _via parameters_, namely via the [automatic `$input` variable](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Automatic_Variables#input), but they wouldn't expect to access pipeline input via _parameters_.

* Technical: ?

**Comment by dbaileyut** [comment]
Yeah, I realized after I posted the conciseness was the same with the alias and that's what I typically use. Also, thanks, I did not know the name for delay-bind script blocks and had trouble finding what made them possible in what seems to be the canonical example, Rename-Item.

This helps me clarify what I really want: a way to use delay-bind script blocks for more (all?) parameters. When I suggested

`Get-Thing | Set-OtherThing -UnsupportedParam %{$_}`

I implicitly wanted the following type of thing to be possible:

`Get-Thing | Set-OtherThing -UnsupportedParam %{ $_.Property -replace 'a','z' }`

I realize that can be written:

`Get-Thing | % { Set-OtherThing -UnsupportedParam ($_.Property -replace 'a','z ) }`

That's not much longer but the post-binding script block type style is just more appealing to me on the command line.
